packages/node/src/login/oidc/ClientRegistrar.spec.ts

```diff
@@ -139,13 +139,58 @@ describe("ClientRegistrar", () => {
 
       // Check that the client information have been saved in storage
       await expect(
-        mockStorage.getForUser("mySession", "clientId", { secure: true })
+        mockStorage.getForUser("mySession", "clientId")
       ).resolves.toEqual(mockDefaultClientConfig().client_id);
       await expect(
-        mockStorage.getForUser("mySession", "clientSecret", { secure: true })
+        mockStorage.getForUser("mySession", "clientSecret")
       ).resolves.toEqual(mockDefaultClientConfig().client_secret);
     });
 
+    it("retrieves client information from storage after dynamic registration", async () => {
+      // Sets up the mock-up for DCR
+      const { Issuer } = jest.requireMock("openid-client");
+      const mockedIssuer = {
+        metadata: mockDefaultIssuerMetadata(),
+        Client: {
+          register: jest
+            .fn()
+            .mockResolvedValueOnce({
+              metadata: mockDefaultClientConfig(),
+            })
+            .mockRejectedValue("Cannot register more than once"),
+        },
+      };
+      Issuer.mockReturnValue(mockedIssuer);
+      const mockStorage = mockStorageUtility({});
+
+      // Run the test
+      const clientRegistrar = getClientRegistrar({
+        storage: mockStorage,
+      });
+      let client = await clientRegistrar.getClient(
+        {
+          sessionId: "mySession",
+          redirectUrl: "https://example.com",
+        },
+        {
+          ...IssuerConfigFetcherFetchConfigResponse,
+        }
+      );
+      // Re-request for the client. If not returned from memory, the mock throws.
```

Comment:
Comment says the mock throws, but I don't see any `expect().toThrow();` to test for an exception being thrown...? And shouldn't it be testing for the expected reason for that exception too (i.e. the reasons being `Cannot register more than once`)...?

Reply:
The `openid-client` mock throws on purpose, to make sure that it isn't called twice, but we never expect the error to be thrown. If the test completes, it means that the client has been retrieved from storage, as intended, because otherwise the error would have been thrown.

Reply:
Oh! Well then I'd rewrite the comment to be clearer, e.g.:
```
// Re-request for the client (which we expect to be returned from memory, as `openid-client` deliberately throws if we attempt to call it twice (since that would indeed be a mistake because...???).
```
(I'm not sure exactly what the 'because ...' part would say, but I know you know :) !)

Reply:
Just to be sure of the content of the comment before re-writing it, `openid-client` does not throw on purpose if we try to register a client twice, but the expected behaviour is to retrieve the already registered client from memory. The mock throws just to make sure that we don't make a second call, but that's not how the actual `openid-client` behaves. What do you think of the following comment: 
```
// Re-request the client information. The client being registered on the first call, the registration information are expected to be retrieved from storage on the second `getClient` call. The `openid-client` mock throws if it is called a second time to ensure that the client isn't registered again.
```

Reply:
Perfect (apart from two teeny weeny tweaks :) !):
```
// Re-request the client information. The client being registered on the first call, the registration information is expected to be retrieved from storage on the second `getClient()` call. The `openid-client` mock throws if it is called a second time to ensure that the client isn't registered again.
```